library/src/main/java/org/mustangproject/ZUGFeRD/ZUGFeRD2PullProvider.java

```diff
@@ -448,10 +448,17 @@ public void generateXML(IExportableTransaction trans) {
 							"</ram:AdditionalReferencedDocument>";
 					}
 				}
-				if (currentItem.getBuyerOrderReferencedDocumentLineID() != null) {
-					xml += "<ram:BuyerOrderReferencedDocument> "
-						+ "<ram:LineID>" + XMLTools.encodeXML(currentItem.getBuyerOrderReferencedDocumentLineID()) + "</ram:LineID>"
-						+ "</ram:BuyerOrderReferencedDocument>";
+				if ((currentItem.getBuyerOrderReferencedDocumentLineID() != null) || (currentItem.getBuyerOrderReferencedDocumentID() != null)) {
+					xml += "<ram:BuyerOrderReferencedDocument> ";
```

Comment:
As a general comment: wouldn't it make sense to use `StringBuilder` in this class instead of creating a new `String` object with each step?

Reply:
yes stringbuilder would make sense